src/apps/origin-dollar-governance/ethereum/origin-dollar-governance.ve-ogv.contract-position-fetcher.ts

```diff
@@ -34,25 +38,45 @@ export class EthereumOriginDollarGovernanceRewardsContractPositionFetcher extend
     return [{ address: '0x0c4576ca1c365868e162554af8e385dc3e7c66d9' }];
   }
 
-  async getTokenDefinitions(): Promise<UnderlyingTokenDefinition[] | null> {
+  async getTokenDefinitions({
+    contract,
+  }: GetTokenDefinitionsParams<Veogv>): Promise<UnderlyingTokenDefinition[] | null> {
+    const address = await contract.ogv();
     return [
       {
-        address: '0x9c354503c38481a7a7a51629142963f98ecc12d0',
+        address,
+        metaType: MetaType.LOCKED,
+        network: this.network,
+      },
+      {
+        address,
         metaType: MetaType.CLAIMABLE,
         network: this.network,
       },
     ];
   }
 
-  async getLabel() {
-    return `Origin Dollar Governance Staking Rewards`;
+  async getLabel({ contractPosition }: GetDisplayPropsParams<Veogv>) {
+    return getLabelFromToken(contractPosition.tokens[0]);
   }
 
   async getTokenBalancesPerPosition({
     address,
     contract,
   }: GetTokenBalancesParams<Veogv, DefaultDataProps>): Promise<BigNumberish[]> {
+    const lockupsRaw = await Promise.all(
+      range(0, 15).map(async index => {
```

Comment:
It might still underreport the net worth if someone has staked more than 15 times. But I'm not sure if anyone has done that many lockups so far. This should be good for the time being

Reply:
It's not perfect as you said someone can have staked more than 15 times and it would underreport. It's a temporary solution until we have the infra to know exactly how many lockups a wallet has without relying on external data or using a subgraph. I will bump it to 30 just in case